uni/lib/view/library/widgets/library_reservations_tab.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni/model/entities/library_reservation.dart';
+import 'package:uni/model/providers/lazy/library_reservations_provider.dart';
+import 'package:uni/view/lazy_consumer.dart';
+import 'package:uni/view/library/widgets/reservation_row.dart';
+
+class LibraryReservationsTab extends StatelessWidget {
+  const LibraryReservationsTab({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return LazyConsumer<LibraryReservationsProvider, List<LibraryReservation>>(
+      builder: (context, reservations) {
+        return LibraryReservationsTabView(reservations);
+      },
+      contentLoadingWidget: const Center(child: CircularProgressIndicator()),
+      hasContent: (reservations) => reservations.isNotEmpty,
+      onNullContent: Center(
+        child: Text(
+          S.of(context).no_data,
+          style: const TextStyle(fontSize: 18),
+        ),
+      ),
+    );
+  }
+
+  Future<void> refresh(BuildContext context) async {
+    await Provider.of<LibraryReservationsProvider>(context, listen: false)
+        .forceRefresh(context);
+  }
+}
+
+class LibraryReservationsTabView extends StatelessWidget {
+  const LibraryReservationsTabView(this.reservations, {super.key});
+  final List<LibraryReservation>? reservations;
+
+  @override
+  Widget build(BuildContext context) {
+    if (reservations == null || reservations!.isEmpty) {
```

Comment:
Wouldn't `LazyConsumer` handle this already?

Reply:
Yes, it is outdated